### PR TITLE
Allow to override mediawikiHost and port with environment variables

### PR DESCRIPTION
**Pull request by rogatty**
This is needed for https://wikia-inc.atlassian.net/browse/OPS-5582

### CODE REVIEW

File: config/localSettings.base.ts [hunk 1]
```diff
@@ -25,7 +25,7 @@ var localSettings: LocalSettings = {
 	},
 	mediawikiHost: 'your-devbox-name',
 	maxRequestsPerChild: parseInt(process.env.MAX_REQUEST_PER_CHILD, 10) || 50000,
-	port: 8000,
+	port: process.env.PORT || 8000,
```
Comment (hakubo) [inline config/localSettings.base.ts]: Shouldn't that be something like MERCURY_PORT?

Comment (rogatty) [inline config/localSettings.base.ts]: Thought about it but there are no prefixes in https://github.com/Wikia/chef-repo/blob/master/cookbooks/_mercury/templates/default/sv-mercury-run.erb

Comment (hakubo) [inline config/localSettings.base.ts]: mkay, if @adamw-wikia is ok with it, I'm am too :)

Comment (hakubo) [inline config/localSettings.base.ts]: @adamw-wikia are you ok with it?

**Comment by rogatty** [comment]
@hakubo @adamw-wikia can you take a look?

**Comment by rogatty** [comment]
There will be some minor conflicts when merging with https://github.com/Wikia/mercury/pull/553 but they should be easy to resolve.

File: config/localSettings.base.ts [hunk 2]
```diff
@@ -23,7 +23,8 @@ var localSettings: LocalSettings = {
 	loggers: {
 		syslog: 'debug'
 	},
-	mediawikiHost: 'your-devbox-name',
+	// This should always be overriden by environment variable or localSettings.ts
+	mediawikiHost: Utils.stripDevboxDomain(process.env.HOST || 'OVERRIDE_ME'),
```
Comment (hakubo) [inline config/localSettings.base.ts]: Not sure about that
there should be something like: process.env.LOGNAME that on my devbox resolves to:
`jolek`

maybe that is a good enough fallback?
Should work on devboxes, 
I also have setup user name on my computer that matches my devbox user,
I think there are more people with such setup

**Comment by hakubo** [comment]
so it works for you as well? nice
thanks!

**Comment by owend** [comment]
553 is ready to merge.  It's been tested on sandbox.  Who gets to resolve the conflicts?  :)

**Comment by rogatty** [comment]
@owend I'll resolve conflicts tomorrow, thanks.

**Comment by hakubo** [comment]
LGTM

**Pull request merged**